cmd/generic-handlers.go

```diff
@@ -532,18 +532,17 @@ func setRequestValidityHandler(h http.Handler) http.Handler {
 	})
 }
 
-var fwd = handlers.NewForwarder(&handlers.Forwarder{
-	PassHost:     true,
-	RoundTripper: newGatewayHTTPTransport(1 * time.Hour),
-	Logger: func(err error) {
-		logger.LogIf(GlobalContext, err)
-	},
-})
-
 // setBucketForwardingHandler middleware forwards the path style requests
 // on a bucket to the right bucket location, bucket to IP configuration
 // is obtained from centralized etcd configuration service.
 func setBucketForwardingHandler(h http.Handler) http.Handler {
+	fwd := handlers.NewForwarder(&handlers.Forwarder{
```

Comment:
You can actually initialize fwd below, in the returned function, after the if check, because we don't want to initialize  fwd for **every** request.

Reply:
But it's not initialized multiple times @vadmeste setBucketForward is only once

Reply:
> But it's not initialized multiple times @vadmeste setBucketForward is only once

setBucketForward() is called in all requests, you can add a log and check. I believe one change changed how we pass handlers to gorilla mux so the concept is changed now.

Reply:
Yes saw it @vadmeste